File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int i = 0, j = nums.length - 1, middle;
        while( i <= j) {
            middle = i + (j - i) / 2;
            if (nums[middle] > target) j = middle - 1;
            else if (nums[middle] < target) i = middle + 1;
            else if (nums[middle] == target) return middle;
        }
    return -1;
    }
}

File: 0704-binary-search/0704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var search = function(nums, target) {
    var l = 0;
    var r = nums.length - 1;

    while (l <= r) {
        let mid = Math.floor((r + l) / 2);
        if (nums[mid] > target) {
            r = mid - 1;
        } else if (nums[mid] < target) {
            l = mid + 1;
        } else {
            return mid;
        }
    };
    return -1; 

};